File: java/src/main/java/com/showmeyourcode/playground/java/code/Samples.java
package com.showmeyourcode.playground.java.code;

import com.showmeyourcode.playground.java.code.paradigm.functional.FunctionalProgramming;
import com.showmeyourcode.playground.java.code.pattern.behavioral.BehavioralDesignPatterns;
import com.showmeyourcode.playground.java.code.pattern.creational.CreationalDesignPatterns;
import com.showmeyourcode.playground.java.code.pattern.structural.StructuralDesignPatterns;
import com.showmeyourcode.playground.java.code.task.Tasks;
import com.showmeyourcode.playground.kotlin.common.Descriptions;

import static com.showmeyourcode.playground.java.LanguageOverview.LOGGER;

public class Samples {

    private Samples(){}

    public static void main(){
        LOGGER.info(Descriptions.header(Descriptions.CODE_SAMPLES));
        Arrays.main();
        Lists.main();
        Classes.main();
        Functions.main();
        Futures.main();
        Concurrency.main();
        LOGGER.info(Descriptions.header(Descriptions.PROGRAMMING_PARADIGMS));
        LOGGER.info(Descriptions.PROGRAMMING_PARADIGM);
        FunctionalProgramming.main();
        LOGGER.info(Descriptions.header(Descriptions.DESIGN_PATTERNS));
        CreationalDesignPatterns.main();
        StructuralDesignPatterns.main();
        BehavioralDesignPatterns.main();

        Tasks.main();
    }
}

File: kotlin/src/main/kotlin/com/showmeyourcode/playground/kotlin/code/Samples.kt
package com.showmeyourcode.playground.kotlin.code

import com.showmeyourcode.playground.kotlin.code.paradigm.functional.FunctionalProgramming
import com.showmeyourcode.playground.kotlin.common.Descriptions
import com.showmeyourcode.playground.kotlin.common.Logging

@Suppress("UNUSED_PARAMETER")
fun main(args: Array<String>) {
    Logging.LOGGER.info("\n\n${Descriptions.headerTemplate(Descriptions.CODE_SAMPLES)}")
    Classes.main()
    Ranges.main()
    Arrays.main()
    Lists.main()
    Coroutines.main()
    Logging.LOGGER.info(Descriptions.headerTemplate(Descriptions.PROGRAMMING_PARADIGMS))
    Logging.LOGGER.info(Descriptions.PROGRAMMING_PARADIGM)
    FunctionalProgramming.main()
}
